Contributor Docs Improvements Round 2 (#18667)
directus/directus: docs/contributing/codebase-overview.md

| `/utils`       | Utility functions used in various parts of the app.                                                                                                          |
| `/views`       | The (two) main views used in the app: public / private. Also contains "internal" coupled components for those two views.                                     |

::: tip Component Library

Directus comes shipped with it's own [Vue Component Library and Storybook](<(https://components.directus.io)>) that you
can use to enrich your extensions or when developing locally. These components can be used in any of the "app
extensions", including Interfaces, Displays, Modules, Layouts, and Panels.

:::

## Additional Packages (`/packages`)

The various sub-packages of the platform. Including the file-storage adapters, schema, specs, etc.

> | Package Name                                                                                                             | Description                                                                                                                      |

Reviewer: Thanks for updating this table! However I wonder if we can make this a little different / more dynamic so we don't have to update this table every time something changes about the packages (which I'm also sure we will forge 🙈😄).

Reviewer 2: Yes, that would be the goal for the TypeDocs. However, in the meantime this will need to be maintained until then. With us getting into a release cycle and low amount of PRs it is inconsequential to keep this updated and it is something that I can make sure stays updated in the meantime 😀

| [@directus/tsconfig](https://github.com/directus/directus/tree/main/packages/tsconfig)                                   | The shared TS Config files used by the projects in the Directus ecosystem.                                                       |
| [@directus/types](https://github.com/directus/directus/tree/main/packages/types)                                         | Shared types for Directus                                                                                                        |
| [@directus/update-check](https://github.com/directus/directus/tree/main/packages/update-check)                           | Check if an update is available for a given package                                                                              |
| [@directus/utils](https://github.com/directus/directus/tree/main/packages/utils)                                         | Utilities shared between the Directus packages                                                                                   |

## Tests (`/tests`)

> Tests are maintained on a per-package base. This folder contains the platform-wide (end-to-end) tests like
> [Blackbox](https://github.com/directus/directus/tree/main/tests/blackbox).

Reviewer: I think I'd prefer to link to the "Tests" documentation page instead. Something like:

```suggestion
Tests are maintained on a per-package base. This folder contains the platform-wide (end-to-end) tests. See [Tests](/contributing/tests) for more information.
```
